Riot/Utils/UniversalLink.m

```diff
@@ -15,21 +15,134 @@
+{
+    NSArray<NSString*> *pathParams;
+    NSMutableDictionary *queryParams = [NSMutableDictionary dictionary];
+
+    NSArray<NSString*> *fragments = [_url.fragment componentsSeparatedByString:@"?"];
+
+    // Extract path params
+    pathParams = [fragments[0] componentsSeparatedByString:@"/"];
+
+    // Remove the first empty path param string
+    pathParams = [pathParams filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"length > 0"]];
+
+    // URL decode each path param
+    pathParams = [pathParams vc_map:^id _Nonnull(NSString * _Nonnull item) {
+        return [item stringByRemovingPercentEncoding];
+    }];
+    
+    // Extract query params
+    NSURLComponents *components = [NSURLComponents componentsWithURL:_url resolvingAgainstBaseURL:NO];
+    for (NSURLQueryItem *item in components.queryItems)
+    {
+        if (item.value)
+        {
+            NSString *key = item.name;
+            NSString *value = item.value;
+            value = [value stringByReplacingOccurrencesOfString:@"+" withString:@" "];
+            value = [value stringByRemovingPercentEncoding];
+
+            if ([key isEqualToString:@"via"])
+            {
+                // Special case the via parameter
+                // As we can have several of them, store each value into an array
+                if (!queryParams[key])
+                {
+                    queryParams[key] = [NSMutableArray array];
+                }
+
+                [queryParams[key] addObject:value];
+            }
+            else
+            {
+                queryParams[key] = value;
+            }
+        }
+    }
+    // Query params are in the form [queryParam1Key]=[queryParam1Value], so the
+    // presence of at least one '=' character is mandatory
+    if (fragments.count == 2 && (NSNotFound != [fragments[1] rangeOfString:@"="].location))
+    {
+        for (NSString *keyValue in [fragments[1] componentsSeparatedByString:@"&"])
+        {
+            // Get the parameter name
+            NSString *key = [keyValue componentsSeparatedByString:@"="][0];
+
+            // Get the parameter value
+            NSString *value = [keyValue componentsSeparatedByString:@"="][1];
+            if (value.length)
+            {
+                value = [value stringByReplacingOccurrencesOfString:@"+" withString:@" "];
+                value = [value stringByRemovingPercentEncoding];
+
+                if ([key isEqualToString:@"via"])
+                {
+                    // Special case the via parameter
+                    // As we can have several of them, store each value into an array
+                    if (!queryParams[key])
+                    {
+                        queryParams[key] = [NSMutableArray array];
+                    }
+
+                    [queryParams[key] addObject:value];
+                }
+                else
+                {
+                    queryParams[key] = value;
+                }
+            }
+        }
+    }
```

Comment:
Is this the old version of the `NSURLComponents` version above? I might have missing something here though 🤔